src/components/LogViewer/index.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import './index.scss';
+
+import React, { useEffect, useRef, useState } from 'react';
+import { connect } from 'react-redux';
+
+import { AppState } from '../../store';
+import { LogEntry, LogState } from '../../store/logviewer';
+import { CompiledGetterCache } from './CompiledGetterCache';
+import History from './History';
+import { QueryBar, parseQuery } from './QueryBar';
+import { ReconnectingWebSocket } from './ReconnectingWebSocket';
+
+interface PropsFromState {
+  log: LogState;
+}
+
+interface PropsFromDispatch {
+  logReceived: Function;
+}
+
+export type Props = PropsFromState & PropsFromDispatch;
+
+const webSocket = new ReconnectingWebSocket();
+const compiledGetters = new CompiledGetterCache();
+
+export const LogViewer: React.FC<Props> = ({ logReceived }) => {
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    webSocket.listener = (log: LogEntry) => {
+      //todo: remove hack to cut off icon
+      log.message = log.message
+        .split(' ')
+        .slice(1)
+        .join(' ');
+      logReceived(log);
+    };
+  }, []);
+
+  const parsedQuery = parseQuery(query);
+
+  return (
+    <div
+      id="log"
+      className="mdc-layout-grid__cell mdc-layout-grid__cell--span-12 mdc-elevation--z2"
```

Comment:
These should be rmwc `<GridCell>` and `<Elevation wrap>`, but I can do this in a polish pass if that's fine with you.

Reply:
sgtm, thanks